src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java

```diff
@@ -405,4 +405,14 @@ public MethodField makeMethodField(String name, List<SQLExpr> arguments, SQLAggr
             return new MethodField(name, paramers, option, alias);
         }
     }
+
+    public ScriptMethodField makeCastMethodField(SQLCastExpr castExpr) throws SqlParseException {
```

Comment:
I wonder what's the intention of this method because I noticed it seems already be covered by `makeField`: https://github.com/opendistro-for-elasticsearch/sql/blob/d3e6477b6e9bea8acf498ae8ca23bdfa69f3c257/src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java#L121

The only difference I notice is that boolean passed to `CastParser.parse()` method. Is all the code change in `WhereParser` due to this new method? Any chance to avoid those changes and reuse the logic in `makeField` above?

Reply:
Yes, that is all the code change in `WhereParser`. I think it is possible but it would require adding a new parameter to `makeField` and `makeFieldImpl` to account for the changed boolean passed. Since those methods are used in many different places and this case is only needed for this, I chose to make a new method, but can refactor if it's deemed the better option.

Reply:
I may need to look into the code but I'm wondering how other functions handle this case of being called in SELECT and WHERE. I can see a boolean parameter in `SQLFunctions.function()` and called by `makeField` with different value.

Reply:
I will step through the code with some other functions to see if there's a way for me to eliminate the need for that method